# Lab2- Max Burkhart

In [19]:
import warnings
warnings.simplefilter('ignore')

import pandas as pd

df = pd.read_csv('s3://ia340-2025-fall-86358/Crime_Data_from_2020_to_Present.csv') # load the data from s3
df.head() #show the top 5 rows

DR_NO               Date Rptd                DATE OCC  TIME OCC  AREA  \
0  211507896  04/11/2021 12:00:00 AM  11/07/2020 12:00:00 AM       845    15   
1  201516622  10/21/2020 12:00:00 AM  10/18/2020 12:00:00 AM      1845    15   
2  240913563  12/10/2024 12:00:00 AM  10/30/2020 12:00:00 AM      1240     9   
3  210704711  12/24/2020 12:00:00 AM  12/24/2020 12:00:00 AM      1310     7   
4  201418201  10/03/2020 12:00:00 AM  09/29/2020 12:00:00 AM      1830    14   

     AREA NAME  Rpt Dist No  Part 1-2  Crm Cd  \
0  N Hollywood         1502         2     354   
1  N Hollywood         1521         1     230   
2     Van Nuys          933         2     354   
3     Wilshire          782         1     331   
4      Pacific         1454         1     420   

                                         Crm Cd Desc  ... Status  Status Desc  \
0                                  THEFT OF IDENTITY  ...     IC  Invest Cont   
1     ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT  ...     IC  Invest Cont   
2                                  THEFT OF IDENTITY  ...     IC  Invest Cont   
3  THEFT FROM MOTOR VEHICLE - GRAND ($950.01 AND ...  ...     IC  Invest Cont   
4    THEFT FROM MOTOR VEHICLE - PETTY ($950 & UNDER)  ...     IC  Invest Cont   

  Crm Cd 1 Crm Cd 2  Crm Cd 3 Crm Cd 4  \
0    354.0      NaN       NaN      NaN   
1    230.0      NaN       NaN      NaN   
2    354.0      NaN       NaN      NaN   
3    331.0      NaN       NaN      NaN   
4    420.0      NaN       NaN      NaN   

                                   LOCATION Cross Street      LAT       LON  
0   7800    BEEMAN                       AV          NaN  34.2124 -118.4092  
1           ATOLL                        AV     N  GAULT  34.1993 -118.4203  
2  14600    SYLVAN                       ST          NaN  34.1847 -118.4509  
3   6000    COMEY                        AV          NaN  34.0339 -118.3747  
4                   4700    LA VILLA MARINA          NaN  33.9813 -118.4350  

[5 rows x 28 columns]

In [27]:
df.shape #Shows rows and columns 
df.info() # Gives a summary of the data
df.describe() # Gives summary statistics such as count, mean, and quartiles

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1004991 entries, 0 to 1004990
Data columns (total 28 columns):
 #   Column          Non-Null Count    Dtype  
---  ------          --------------    -----  
 0   DR_NO           1004991 non-null  int64  
 1   Date Rptd       1004991 non-null  object 
 2   DATE OCC        1004991 non-null  object 
 3   TIME OCC        1004991 non-null  int64  
 4   AREA            1004991 non-null  int64  
 5   AREA NAME       1004991 non-null  object 
 6   Rpt Dist No     1004991 non-null  int64  
 7   Part 1-2        1004991 non-null  int64  
 8   Crm Cd          1004991 non-null  int64  
 9   Crm Cd Desc     1004991 non-null  object 
 10  Mocodes         853372 non-null   object 
 11  Vict Age        1004991 non-null  int64  
 12  Vict Sex        860347 non-null   object 
 13  Vict Descent    860335 non-null   object 
 14  Premis Cd       1004975 non-null  float64
 15  Premis Desc     1004403 non-null  object 
 16  Weapon Used Cd  327247 non-null   fl

DR_NO      TIME OCC          AREA   Rpt Dist No      Part 1-2  \
count  1.004991e+06  1.004991e+06  1.004991e+06  1.004991e+06  1.004991e+06   
mean   2.202215e+08  1.339900e+03  1.069174e+01  1.115633e+03  1.400348e+00   
std    1.319718e+07  6.510613e+02  6.110255e+00  6.111605e+02  4.899691e-01   
min    8.170000e+02  1.000000e+00  1.000000e+00  1.010000e+02  1.000000e+00   
25%    2.106169e+08  9.000000e+02  5.000000e+00  5.870000e+02  1.000000e+00   
50%    2.209159e+08  1.420000e+03  1.100000e+01  1.139000e+03  1.000000e+00   
75%    2.311103e+08  1.900000e+03  1.600000e+01  1.613000e+03  2.000000e+00   
max    2.521041e+08  2.359000e+03  2.100000e+01  2.199000e+03  2.000000e+00   

             Crm Cd      Vict Age     Premis Cd  Weapon Used Cd      Crm Cd 1  \
count  1.004991e+06  1.004991e+06  1.004975e+06   327247.000000  1.004980e+06   
mean   5.001568e+02  2.891706e+01  3.056201e+02      363.955300  4.999174e+02   
std    2.052731e+02  2.199272e+01  2.193021e+02      123.734528  2.050736e+02   
min    1.100000e+02 -4.000000e+00  1.010000e+02      101.000000  1.100000e+02   
25%    3.310000e+02  0.000000e+00  1.010000e+02      311.000000  3.310000e+02   
50%    4.420000e+02  3.000000e+01  2.030000e+02      400.000000  4.420000e+02   
75%    6.260000e+02  4.400000e+01  5.010000e+02      400.000000  6.260000e+02   
max    9.560000e+02  1.200000e+02  9.760000e+02      516.000000  9.560000e+02   

           Crm Cd 2     Crm Cd 3   Crm Cd 4           LAT           LON  
count  69160.000000  2314.000000   64.00000  1.004991e+06  1.004991e+06  
mean     958.101258   984.015990  991.21875  3.399821e+01 -1.180909e+02  
std      110.354348    52.350982   27.06985  1.610713e+00  5.582386e+00  
min      210.000000   310.000000  821.00000  0.000000e+00 -1.186676e+02  
25%      998.000000   998.000000  998.00000  3.401470e+01 -1.184305e+02  
50%      998.000000   998.000000  998.00000  3.405890e+01 -1.183225e+02  
75%      998.000000   998.000000  998.00000  3.416490e+01 -1.182739e+02  
max      999.000000   999.000000  999.00000  3.433430e+01  0.000000e+00

######## This data has 27 columns and 1,004,991 rows. There are three types of data entries: float64, object, and int64. "float64" are numbers with decimals such as 3.14. "int64" are whole numbers such as 42. "objects" are text such as apple. My first impression is that Los Angeles is crime-ridden city, with over a million recorded crimes in this dataset alone!

In [24]:
df.shape #Show the number of rows and columns
df['DR_NO'] #Select a column
df.sort_values (by='Crm Cd Desc') #sort data by column

DR_NO               Date Rptd                DATE OCC  TIME OCC  \
709797  230515242  10/30/2023 12:00:00 AM  10/30/2023 12:00:00 AM       800   
282575  210108514  04/07/2021 12:00:00 AM  04/07/2021 12:00:00 AM       839   
568531  221407662  03/16/2022 12:00:00 AM  03/15/2022 12:00:00 AM      2100   
770494  231507203  03/18/2023 12:00:00 AM  03/17/2023 12:00:00 AM       255   
483207  220315512  08/22/2022 12:00:00 AM  08/20/2022 12:00:00 AM      2315   
...           ...                     ...                     ...       ...   
622027  220717413  12/05/2022 12:00:00 AM  12/05/2022 12:00:00 AM      1210   
164613  201216692  07/18/2020 12:00:00 AM  07/18/2020 12:00:00 AM        25   
84259   202006611  01/17/2020 12:00:00 AM  01/16/2020 12:00:00 AM      2000   
110727  200808185  04/10/2020 12:00:00 AM  04/06/2020 12:00:00 AM      2030   
666840  240604490  01/11/2024 12:00:00 AM  11/30/2023 12:00:00 AM       700   

        AREA    AREA NAME  Rpt Dist No  Part 1-2  Crm Cd  \
709797     5       Harbor          566         1     648   
282575     1      Central          111         1     648   
568531    14      Pacific         1412         1     648   
770494    15  N Hollywood         1526         1     648   
483207     3    Southwest          334         1     648   
...      ...          ...          ...       ...     ...   
622027     7     Wilshire          721         2     756   
164613    12  77th Street         1248         2     756   
84259     20      Olympic         2088         2     756   
110727     8      West LA          833         2     756   
666840     6    Hollywood          668         2     756   

                       Crm Cd Desc  ... Status   Status Desc Crm Cd 1  \
709797                       ARSON  ...     IC   Invest Cont    648.0   
282575                       ARSON  ...     IC   Invest Cont    648.0   
568531                       ARSON  ...     IC   Invest Cont    648.0   
770494                       ARSON  ...     IC   Invest Cont    648.0   
483207                       ARSON  ...     AA  Adult Arrest    648.0   
...                            ...  ...    ...           ...      ...   
622027  WEAPONS POSSESSION/BOMBING  ...     IC   Invest Cont    756.0   
164613  WEAPONS POSSESSION/BOMBING  ...     IC   Invest Cont    756.0   
84259   WEAPONS POSSESSION/BOMBING  ...     IC   Invest Cont    756.0   
110727  WEAPONS POSSESSION/BOMBING  ...     IC   Invest Cont    756.0   
666840  WEAPONS POSSESSION/BOMBING  ...     AA  Adult Arrest    756.0   

       Crm Cd 2  Crm Cd 3 Crm Cd 4                                 LOCATION  \
709797      NaN       NaN      NaN  1100 S  PALOS VERDES                 ST   
282575      NaN       NaN      NaN   600 N  SPRING                       ST   
568531      NaN       NaN      NaN          WESTMINSTER                  AV   
770494      NaN       NaN      NaN  6200    BECK                         AV   
483207      NaN       NaN      NaN  3000    CRENSHAW                     BL   
...         ...       ...      ...                                      ...   
622027    998.0       NaN      NaN  8700    GRACIE ALLEN                 DR   
164613    998.0       NaN      NaN   900 W  64TH                         ST   
84259       NaN       NaN      NaN       W  WASHINGTON                   BL   
110727      NaN       NaN      NaN  1300    MIDVALE                      AV   
666840      NaN       NaN      NaN                               ST ANDREWS   

                           Cross Street      LAT       LON  
709797                              NaN  33.7342 -118.2814  
282575                              NaN  34.0582 -118.2387  
568531                             MAIN  33.9901 -118.4726  
770494                              NaN  34.1830 -118.3834  
483207                              NaN  34.0276 -118.3351  
...                                 ...      ...       ...  
622027                              NaN  34.0753 -118.3795  
164613                      

In [30]:
df=['Crm Cd Desc']= 'Type of Crime'

SyntaxError: cannot assign to literal (2795685026.py, line 1)